architxt/database/__init__.py

```diff
@@ -127,6 +134,35 @@ def read_table(
         yield Tree("ROOT", children)
 
 
+def read_unreferenced_table(parent_table: Table, foreign_table: ForeignKey, *, conn: Connection, sample: int = 0, visited_links: set[ForeignKey] = set()) -> Generator[Tree, None, None]:
```

Comment:
Do not use a mutable object as a default parameter value see [Python Documentation](https://docs.python.org/3/tutorial/controlflow.html#default-argument-values) for why. In this case, it seems unnecessary since the table is now considered the root, meaning we should follow every link and reset it for all unreferenced rows. As evidence, the value is never actually used apart from being set.